### PR TITLE
Feat/auth middleware

### PR DESCRIPTION
**Pull request by tobi-bams**

### CODE REVIEW

File: src/server/middlewares/auth.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  LoggerInstance.info("Verifying Customer Auth token: %o")
+  try {
+    const verify: VerifyTokenRes = verifyToken(req)
+    if (!verify.success) {
+      return res.status(401).json({ success: false, message: verify.msg })
+    }
+    if (verify.payload?.admin) {
+      return res.status(401).json({ success: false, message: "unauthorize" })
+    }
+    const user = (await Account.findOne({
```
Comment (Extheoisah) [inline src/server/middlewares/auth.ts:39]: finding an account should be the service layer. You can create an `AccountRepository` function in `services/Account` that receives an `id` and returns the user account.

File: src/server/middlewares/auth.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  LoggerInstance.info("Verifying Admin Auth token: %o")
+  try {
+    const verify: VerifyTokenRes = verifyToken(req)
+    if (!verify.success) {
+      return res.status(401).json({ success: false, message: verify.msg })
+    }
+    if (!verify.payload?.admin) {
+      return res.status(401).json({ success: false, message: "unauthorize" })
+    }
+    const user = (await Account.findOne({
```
Comment (Extheoisah) [inline src/server/middlewares/auth.ts:63]: same here. `AccountRepositoty`

**Review by Extheoisah — CHANGES_REQUESTED** [review]

File: src/domain/shared/primitives.ts [hunk 3]
```diff
@@ -42,3 +42,9 @@ export enum TransactionMetaType {
   ONCHAIN = "ONCHAIN",
   FIAT = "FIAT",
 }
+
+export interface JwtPayload {
```
[the remark this answers is not in the document]
Comment (tobi-bams) [reply src/domain/shared/primitives.ts:46]: could this be in the `domain/authentication` part??

File: src/server/middlewares/auth.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  msg?: string
+  payload?: JwtPayload
+}
+
+function verifyToken(req: Request): VerifyTokenRes {
+  LoggerInstance.info("Verifying Auth token: %o")
+  try {
+    const token = req.headers.authorization?.split(" ")[1]
+    if (!token) {
+      return { success: true, msg: "invalid token" }
```
[the remark this answers is not in the document]
Comment (tobi-bams) [reply src/server/middlewares/auth.ts:19]: Seem's I am not getting you.

I don't want to throw an error, I just want to return an error

**Review by Extheoisah — APPROVED** [review]

**Pull request merged**